### PR TITLE
Migrate many custom matchers to TypeMatcher

### PR DESCRIPTION
**Pull request by kevmoo**
And change the public signature to allow better reuse

### CODE REVIEW

**Comment by kevmoo** [comment]
ignoring changelog change for pkg:file – just added the dependency override for testing

**Comment by kevmoo** [comment]
FYI: just changed return types to be more specific, so shouldn't be a breaking change

File: pkgs/file_testing/lib/src/testing/core_matchers.dart [hunk 1]
```diff
@@ -67,89 +71,10 @@ void expectFileSystemException(dynamic osErrorCode, void Function() callback) {
...
-      return _matcher!.describe(desc);
-    }
-  }
-}
-
-class _HasPath extends Matcher {
-  _HasPath(dynamic path) : _matcher = wrapMatcher(path);
-
-  final Matcher _matcher;
+final TypeMatcher exists =
```
Comment (natebosch) [inline pkgs/file_testing/lib/src/testing/core_matchers.dart:74]: [optional] Would it be worthwhile adding the generic?

```suggestion
final TypeMatcher<FileSystemEntity> exists =
```

**Review by natebosch — APPROVED** [review]

File: pkgs/file_testing/lib/src/testing/core_matchers.dart (hunk 1 above)
Comment (kevmoo) [reply pkgs/file_testing/lib/src/testing/core_matchers.dart]: Good catch! I guess I could have just omitted it and let it get inferred.

**Pull request merged**